internal/settings/linode.go

```diff
@@ -0,0 +1,328 @@
+package settings
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/qdm12/ddns-updater/internal/constants"
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/regex"
+)
+
+type linode struct {
+	domain    string
+	host      string
+	ipVersion models.IPVersion
+	token     string
+}
+
+func NewLinode(data json.RawMessage, domain, host string, ipVersion models.IPVersion,
+	_ bool, _ regex.Matcher) (s Settings, err error) {
+	extraSettings := struct {
+		Token string `json:"token"`
+	}{}
+	if err := json.Unmarshal(data, &extraSettings); err != nil {
+		return nil, err
+	}
+	l := &linode{
+		domain:    domain,
+		host:      host,
+		ipVersion: ipVersion,
+		token:     extraSettings.Token,
+	}
+	if err := l.isValid(); err != nil {
+		return nil, err
+	}
+	return l, nil
+}
+
+func (l *linode) isValid() error {
+	if len(l.token) == 0 {
+		return ErrEmptyToken
+	}
+	return nil
+}
+
+func (l *linode) String() string {
+	return toString(l.domain, l.host, constants.LUADNS, l.ipVersion)
```

Comment:
Should this be Linode?

Reply:
It definitely should 👍  Although that's just a display thing, it won't affect the rest of the code. Thanks for spotting it!